website/client/src/router/index.js

```diff
@@ -47,6 +47,9 @@ const HallPage = () => import(/* webpackChunkName: "hall" */'@/components/hall/i
 const PatronsPage = () => import(/* webpackChunkName: "hall" */'@/components/hall/patrons');
 const HeroesPage = () => import(/* webpackChunkName: "hall" */'@/components/hall/heroes');
 
+// Admin Panel
+const AdminPanelPage = () => import(/* webpackChunkName: "admin-panel" */'@/components/admin-panel');
```

Comment:
I think that other than the menu item being hidden, if someone tries to open the route and they're not allowed they should be redirected somewhere else, see here https://github.com/HabitRPG/habitica/blob/develop/website/client/src/router/index.js#L356-L368 for an example

Reply:
I've tried that in this commit:
https://github.com/HabitRPG/habitica/pull/12035/commits/a4f9c754ad3886466925de4d8ed9a5ac55d2f999
but it needs help as shown in TODO comment near the bottom of the commit.

By the way, I've done it with a `requiresAdmin` true/false option but it could also be done with an integer option called `requiresMinimumContributorTier` which could be set to 8 for the user editing feature (mods and staff) and 9 for the CMS (staff only), with 0 as the default. Let me know if you prefer that! (The TODO comment would still need to be fixed - that's not dependent on which option is used.)

Reply:
Hmm I see, the issue is that the user is not loaded yet at this point, looking at the Vue router docs these navigation guards can return a promise so we can try transforming the function to an async function but to avoid it impacting all routes it should wait for the user to be loaded only when admin privileges are required:

- I would remove `const isAdmin`
- transform the if to something like this

```
if (routeRequiresAdmin) {
    await this.$store.dispatch('user:fetch')
    const isUserAdmin = isUserLoaded && store.state.user.data.contributor
    && store.state.user.data.contributor.admin;

    ....
}
```

Note that I haven't tried the fix

Reply:
That worked! Thanks! It's done here: https://github.com/HabitRPG/habitica/pull/12035/commits/566d9a9a95358073a971f30372398d199b62da64